src/compiler/factory.ts

```diff
@@ -197,6 +197,15 @@ namespace ts {
         return name;
     }
 
+    export function getOptimisticScopedGeneratedNameForName(node: Identifier): Identifier {

```

Comment:
Do we need a special factory function for this? Couldn't we just add the extra flags to `autoGenerateflags` after calling one of the existing factory functions?

Reply:
Probably? But I'd also probably still write that into a function so it stays as an inline call as the use site.

Reply:
I'd rather we just change the signature of `getGeneratedNameForNode` to something like:

`export function getGeneratedNameForNode(node: Node, flags?: GeneratedIdentifierFlags): Identifier`

Also, the existing `getGeneratedNameForNode` actually defers reading the text of the node until emit. This is important so you could correctly handle a case where the node you are generating a name from is itself a generated name. What happens if you need to end up with `<T>(x: T) => <T_1>(y: T_1) => T_1_1`?